### PR TITLE
[MBL-17643][All] Dependency update

### PR DESCRIPTION
**Pull request by tamaskozmer**
Dev notes: Most important changes and most changes are because of the Kotlin major version upgrade and the Kotlin Coroutines upgrade.

Test plan: Smoke test the app. Test these things thoroughly:
- Features that are not covered by many tests or hard to automate
- Submission screen, student submissions, comments, changing attempts etc. There were some migration code changes on that screen.
- Video player
- Push notifications

refs: MBL-17643
affects: All
release note: none

## Checklist

- [x] Run E2E test suite
- [x] Tested in dark mode
- [x] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by tamaskozmer** [comment]
The only build failures are from flaky e2e tests, so this is ready to merge if approved.

File: apps/student/src/test/java/com/instructure/student/test/assignment/details/submissionDetails/SubmissionDetailsEffectHandlerTest.kt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*
  * Copyright (C) 2019 - present Instructure, Inc.
  *
- *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     Licensed under the ApachĮe License, Version 2.0 (the "License");
```
Comment (kristofnemere) [inline apps/student/src/test/java/com/instructure/student/test/assignment/details/submissionDetails/SubmissionDetailsEffectHandlerTest.kt:4]: I guess this wasn't intentional

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/discussion/router/DiscussionRouterViewModelTest.kt [hunk 2]
```diff
@@ -32,7 +32,7 @@ class DiscussionRouterViewModelTest {
     private val lifecycleOwner: LifecycleOwner = mockk(relaxed = true)
     private val lifecycleRegistry = LifecycleRegistry(lifecycleOwner)
 
-    private val testDispatcher = TestCoroutineDispatcher()
+    private val testDispatcher = UnconfinedTestDispatcher()
```
Comment (kristofnemere) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/discussion/router/DiscussionRouterViewModelTest.kt:35]: Wouold be good to add  a tearDown method for this test

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/elementary/importantdates/ImportantDatesViewModelTest.kt [hunk 3]
```diff
@@ -64,7 +64,7 @@ class ImportantDatesViewModelTest {
     private val lifecycleOwner: LifecycleOwner = mockk(relaxed = true)
     private val lifecycleRegistry = LifecycleRegistry(lifecycleOwner)
 
-    private val testDispatcher = TestCoroutineDispatcher()
+    private val testDispatcher = UnconfinedTestDispatcher()
```
Comment (kristofnemere) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/elementary/importantdates/ImportantDatesViewModelTest.kt:67]: Would be good to add Dispatchers.resetMain() to tearDown method

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/notification/preferences/EmailNotificationSettingsViewModelTest.kt [hunk 4]
```diff
@@ -57,7 +57,7 @@ class EmailNotificationPreferencesViewModelTest {
     private val lifecycleOwner: LifecycleOwner = mockk(relaxed = true)
     private val lifecycleRegistry = LifecycleRegistry(lifecycleOwner)
 
-    private val testDispatcher = TestCoroutineDispatcher()
+    private val testDispatcher = UnconfinedTestDispatcher()
```
Comment (kristofnemere) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/notification/preferences/EmailNotificationSettingsViewModelTest.kt:60]: Would be good to add Dispatchers.resetMain() to tearDown method

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/file/upload/FileUploadViewModelTest.kt [hunk 5]
```diff
@@ -51,7 +51,7 @@ class FileUploadViewModelTest {
     private val lifecycleOwner: LifecycleOwner = mockk(relaxed = true)
     private val lifecycleRegistry = LifecycleRegistry(lifecycleOwner)
 
-    private val testDispatcher = TestCoroutineDispatcher()
+    private val testDispatcher = UnconfinedTestDispatcher()
```
Comment (kristofnemere) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/file/upload/FileUploadViewModelTest.kt:54]: Would be good to add Dispatchers.resetMain() to tearDown method

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/elementary/schedule/ScheduleViewModelTest.kt [hunk 6]
```diff
@@ -78,7 +78,7 @@ class ScheduleViewModelTest {
     private val lifecycleOwner: LifecycleOwner = mockk(relaxed = true)
     private val lifecycleRegistry = LifecycleRegistry(lifecycleOwner)
 
-    private val testDispatcher = TestCoroutineDispatcher()
+    private val testDispatcher = UnconfinedTestDispatcher()
```
Comment (kristofnemere) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/elementary/schedule/ScheduleViewModelTest.kt:81]: Would be good to add Dispatchers.resetMain() to tearDown method

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/notification/preferences/PushNotificationPreferencesViewModelTest.kt [hunk 7]
```diff
@@ -53,7 +53,7 @@ class PushNotificationPreferencesViewModelTest {
     private val lifecycleOwner: LifecycleOwner = mockk(relaxed = true)
     private val lifecycleRegistry = LifecycleRegistry(lifecycleOwner)
 
-    private val testDispatcher = TestCoroutineDispatcher()
+    private val testDispatcher = UnconfinedTestDispatcher()
```
Comment (kristofnemere) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/notification/preferences/PushNotificationPreferencesViewModelTest.kt:56]: Would be good to add Dispatchers.resetMain() to tearDown method

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/shareextension/progress/ShareExtensionProgressViewModelTest.kt [hunk 8]
```diff
@@ -41,7 +41,7 @@ class ShareExtensionProgressViewModelTest {
     private val lifecycleOwner: LifecycleOwner = mockk(relaxed = true)
     private val lifecycleRegistry = LifecycleRegistry(lifecycleOwner)
 
-    private val testDispatcher = TestCoroutineDispatcher()
+    private val testDispatcher = UnconfinedTestDispatcher()
```
Comment (kristofnemere) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/shareextension/progress/ShareExtensionProgressViewModelTest.kt:44]: Would be good to add Dispatchers.resetMain() to tearDown method

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/shareextension/target/ShareExtensionTargetViewModelTest.kt [hunk 9]
```diff
@@ -53,7 +53,7 @@ class ShareExtensionTargetViewModelTest {
     private val lifecycleOwner: LifecycleOwner = mockk(relaxed = true)
     private val lifecycleRegistry = LifecycleRegistry(lifecycleOwner)
 
-    private val testDispatcher = TestCoroutineDispatcher()
+    private val testDispatcher = UnconfinedTestDispatcher()
```
Comment (kristofnemere) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/shareextension/target/ShareExtensionTargetViewModelTest.kt:56]: Would be good to add Dispatchers.resetMain() to tearDown method

**Review by kristofnemere — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Review by domonkosadam — APPROVED** [review]

**Comment by domonkosadam** [comment]
QA + 1

**Pull request merged**